File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let minutesPerLayer = 2

let remainingMinutesInOven minutesPassed = expectedMinutesInOven - minutesPassed

let preparationTimeInMinutes layers = layers * minutesPerLayer

let elapsedTimeInMinutes layers minutesInOven = preparationTimeInMinutes layers + minutesInOven

File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven()
    {
        return 40;
    }

    public int RemainingMinutesInOven(int elapsedMinutes)
    {
        return this.ExpectedMinutesInOven() - elapsedMinutes;
    }

    public int PreparationTimeInMinutes(int layers)
    {
        return layers * 2;
    }

    public int ElapsedTimeInMinutes(int layers, int elapsedMinutes)
    {
        return this.PreparationTimeInMinutes(layers) + elapsedMinutes;
    }
}
